### DISCUSSION TITLE
Minimal expecification LTSP Client hardware

### DISCUSSION THREAD

**Discussion by doguibnu** [ltsp/ltsp#169] [Q&A]
Hello!
Some of my clients config:
celeron 430 1,8 GHZ
1 GB Ram. 

My server 
Ryzen 3 2200 with Radeon Vega Graphics x4
8 GB Ram
SSD 120 GB
Linux Mint (Mate) 19.1 64 bits

In my tests with ONE client is very slow to open firefox app and Libre Office 
Please, any suggestion?

Thank you

Douglas
@alkisg

**Comment by alkisg** [comment]
Hi Douglas,

LTSP doesn't really affect the OS speed. Your clients will go as fast as if Linux Mint MATE was installed in a local disk.

Cpubenchmark says that your [client score is 478](https://www.cpubenchmark.net/cpu.php?cpu=Intel+Celeron+430+%40+1.80GHz&id=651), while your [server CPU is 7315](https://www.cpubenchmark.net/cpu.php?cpu=AMD+Ryzen+3+2200G&id=3186).

You would get much better results if you did the opposite, and the client had 7315 score and the server had 478 score! :smile:

Acceptable browsing is around 1000 score, while noone should be buying new clients with less than 3000 score. For example, my new laptop has [12442 score](https://www.cpubenchmark.net/cpu.php?cpu=Intel+Core+i7-8750H+%40+2.20GHz&id=3237).

**Comment by enaut** [comment]
@doguibnu well network performance makes the starting of programms a little slower.

With clients that slow I'd choose a different browser like https://wiki.gnome.org/Apps/Web/ . The same for office. choose a very lightweight office. Then it should work.

**Comment by doguibnu** [comment]
Hello @alkisg and @enaut 
So, I am lost. I am getting old pcs (these celeron model are the most) to be used for a school lab. 
Yes @enaut about network. I will use TP-Link switch 24 ports (10/100, not GIGA because the NIC from old pcs is not GIGA) 

@alkisg  when made my first tests, still using LTSP5 looks like the image client run more faster (firefox, youtube) in the tests and my server was a poor amd phenon II x2. 
Owwww, I am lost. 

Thank you

**Comment by alkisg** [comment]
LTSP5 supports "thin clients". This means that the apps run on the server, and the screen is transferred to the clients, like "remote desktop".

This model worked fine until 10+ years ago, when everyone started wanting "simple web browsing" with multimedia. A simple HD video needs 1920\*1080\*32bpp\*60fps=3,981,312,000 bits per second.
But no existing LAN can offer 4 gbps per client, so it's not a valid model anymore.

It's not just about videos. Even scrolling a libreoffice sheet or the fade-in of an impress presentation generate lots of network traffic. And the xorg developers themselves stopped caring too much about the "network transparency" of X, and remote Xorg became more and more buggy.

So... currently:
You may use LTSP5 with the thin client model. It will work with a couple of clients, but for e.g. 10 clients it will be much worse than LTSP19 (which is the same as local installations or LTSP5 with fat clients).

So the best you can do, is like enaut said, to select as light programs as you can. Don't expect a lot of difference though; it's the web itself that's different nowadays and needs more cpu/ram/gpu, not the browsers themselves.

Note that a switch with even 1 gbps port will be 10 times faster than a 100mbps switch. This is because the server will be sending 1 gbps to 10 clients (=100 each), not 100 mbps to 10 clients (=10 each).

You may also try some remote desktop technologies like x2go or xfreerdp in LTSP19; again, don't expect much.

**Comment by doguibnu** [comment]
Hello Alkis!

Thanks about yours considerations, specifications. thanks for the explanation of the switch, very important. So, I think the best minimal client config at least: 4 GB Ram and dual core cpu or best. Less than that, no good experience. Right?

Thanks your attention and help.


> LTSP5 supports "thin clients". This means that the apps run on the server, and the screen is transferred to the clients, like "remote desktop".
> 
> This model worked fine until 10+ years ago, when everyone started wanting "simple web browsing" with multimedia. A simple HD video needs 1920*1080*32bpp*60fps=3,981,312,000 bits per second.
> But no existing LAN can offer 4 gbps per client, so it's not a valid model anymore.
> 
> It's not just about videos. Even scrolling a libreoffice sheet or the fade-in of an impress presentation generate lots of network traffic. And the xorg developers themselves stopped caring too much about the "network transparency" of X, and remote Xorg became more and more buggy.
> 
> So... currently:
> You may use LTSP5 with the thin client model. It will work with a couple of clients, but for e.g. 10 clients it will be much worse than LTSP19 (which is the same as local installations or LTSP5 with fat clients).
> 
> So the best you can do, is like enaut said, to select as light programs as you can. Don't expect a lot of difference though; it's the web itself that's different nowadays and needs more cpu/ram/gpu, not the browsers themselves.
> 
> Note that a switch with even 1 gbps port will be 10 times faster than a 100mbps switch. This is because the server will be sending 1 gbps to 10 clients (=100 each), not 100 mbps to 10 clients (=10 each).
> 
> You may also try some remote desktop technologies like x2go or xfreerdp in LTSP19; again, don't expect much.

**Comment by alkisg** [comment]
The minimal client config for LTSP is whatever the distribution minimal client config is.
[The Mint site](https://blog.linuxmint.com/?s=System+requirements) mentions "1GB RAM (2GB recommended for a comfortable usage)."

That said, **my** recommendation for any desktop Linux with normal web browsing etc (and it's the same for LTSP desktop Linux) is:

- 2 GB RAM for existing clients; 4+ for new clients.
- 1000 cpubenchmark.net CPU score for existing clients; 3000+ for new clients. Do not count cores. There are fast dual core systems, and there are slow quad core systems.

We do have schools with Pentium 4 clients here, with 512 MB or 1 GB RAM. We don't expect to run web browsers comfortably there, but traditional applications like tuxpaint, tuxtype etc run great.
It's not about the OS nor about LTSP; it's the browsers and the modern web that are heavy. :smile:

Btw, if I were to compare the LTSP speed with a local installation in a bit more detail, I'd say that:

- LTSP apps launch a bit faster than local HDDs due to server-side caching
- But they launch a bit slower than local SSDs
- LTSP also blacklists some services, saving a bit of RAM and CPU
- But LTSP /home/username access is a bit slower than local installations (SSD on the server there helps)

So the overall difference isn't very noticeable, that's why I'm saying "LTSP is similar to local installations wrt requirements and speed".

**Comment by doguibnu** [comment]
Does the OS can do difference?
The OS Q4OS  [https://q4os.org/index.html](url)
It runs very fast on simple hardware. It is good, debian base, easy to use
Does It can help at client side to run better?

Sorry for many questions
Thanks

**Comment by alkisg** [comment]
No, the OS doesn't matter much. Sure, LXDE is lighter than GNOME, but they run firefox or tuxpaint at about the same speed. MATE, LXDE, LXQT, XFCE are all fine choices for little RAM and low graphics specs.

Most browsers are about the same too, it's the web pages that are heavy nowadays.

Desktop apps do make a real difference though. There are light apps that can run fine even with 512 RAM and old CPUs. So if you don't care much about web surfing, then you can use LTSP or local installations even with old hardware.

**Comment by enaut** [comment]
Well both firefox and chrome do load a lot of userprofile stuff. Thats why I recommended less complex browsers. It is the same for everything else. If you want to test something just create a live usb key and try it on one of your client boxes. If it performs well the performance will be similar when run via ltsp netboot.

All in all I noticed that programs take longer to load but are quite fast once started (3GB, dual core). For our usecase this is very acceptable I just tell them to boot the computers while I introduce them to the new topics and tasks.

**Comment by alkisg** [comment]
Note that with the new LTSP, testing a new OS in the clients is easier with LTSP rather than with a live CD.

One can download e.g. some ubuntu.iso and put it to /srv/ltsp/images/x86_64.img, run `ltsp kernel x86_64` and boot the clients. No `ltsp image` involved.

**Comment by doguibnu** [comment]
Hello
Great! @enaut and @alkisg 

Getting notes to have good LTSP client (comfortable):
Is better have clients with at least 3 or 4 GB Ram
CPU at least dual core (and not count the cores) checking cpubenchmark between 1000 and 3000.
SSD
GIGA switch ports
cat6 cable
For Graphics speed choose: xfce, lxde,mate and lxqt

Thank you so much

**Comment by doguibnu** [comment]
I need to correct this text!

PLEASE, NOT, NEVER USE CELERON 1.X GHZ PROCESSOR **NEVER**
I switched at least for pentium 2.x and 3.x or amd athlon x2 (works nice)
Now I have only these old processors. For a new lab we are looking to test raspberry py who knows

Thanks





Disconsider
Hello Friends!
How are you?

I would to tell you that I change the distro, from: Linux MInt (Mate or XFCE) to Xubuntu 18.04.01 64 bits. People: SIMPLY GREAT.
I am running client with celeron 430 1,8 GHZ and 1 GB Ram.
All running good: Firefox (various tabs, nice), Libreoffice at least one client about test. 
So, Mint is not good to client LTSP. I do not know why but, it stops, heavy and not works fine. 
I believe that all *buntu (lxde, xfce, mate, lxqt) should running good. 

Thanks you

Douglas
